tic-tac-toe/src/redux/auth/reducer.js

```diff
@@ -0,0 +1,17 @@
+import { actions } from './actions.js';
+
+export default function reducer1(state = 0, action) {
+  switch (action.type) {
+    case actions.VALIDATE_USER:
+      break;
+
+    case actions.VALIDATE_USER_SUCCESS:
+      state = { auth: { email: action.email, token: action.token } };
```

Comment:
1. You need to `return` this.
2. You need to use the spread operator for this. This branch of the state will be an object with an `auth` property. Inside that `auth` object, you need to add `email` and `token`.